tests/unit/commands/init-test.js

```diff
@@ -200,4 +210,37 @@ describe('init command', function() {
         expect(reason).to.equal('Called run');
       });
   });
+
+  it('Registers blueprint options in beforeRun', function() {
+    stub(Blueprint, 'lookup', function(name) {
+      expect(name).to.equal('app');
+      return {
+        availableOptions: [
+          { name: 'custom-blueprint-option', type: String }
+        ]
+      };
+    }, true);
+
+    buildCommand();
+
+    command.beforeRun(['app']);
+    expect(pluck(command.availableOptions, 'name')).to.contain('custom-blueprint-option');
+  });
+
+  it('Passes command options through to the install blueprint task', function() {
+    tasks.InstallBlueprint = Task.extend({
+      run: function(blueprintOpts) {
+        expect(blueprintOpts).to.contain.keys('customOption');
+        expect(blueprintOpts.customOption).to.equal('customValue');
+        return Promise.reject('Called run');
+      }
+    });
+
+    buildCommand();
+
+    return command.validateAndRun(['--custom-option=customValue'])
+      .catch(function(reason) {
+        expect(reason).to.equal('Called run');
+      });
```

Comment:
this test may pass, if the promise does not reject. This can easily result in false positives, we should likely assert the promise returned from `validateAndRun` always rejects